Slpi test support

wangxdflight opened the pull request.

mcharleb committed:
Added support for SLPI

Signed-off-by: Mark Charlebois <mcharleb@qti.qualcomm.com>

wangxdflight committed:
[ATL-6179] Mainline malloc size test

wangxdflight committed:
[ATL-6180] added test case of file I/O in a seperate thread

wangxdflight committed:
[ATL-6192] mainline SPI loopback and IMU whoami test

wangxdflight committed:
[ATL-6177] added compilation flag for ADSP/SLPI, moved port definitions to a seperate file

wangxdflight committed:
updated cmake_hexagon submodule

wangxdflight committed:
tested well on eagle board

wangxdflight committed:
updated GPIO number for SLPI

wangxdflight committed:
need to set larger pthread stack size for SLPI

mcharleb commented on test/dspal_tester/adsp_proc/misc_tests.c:
```diff
@@ -39,10 +39,10 @@ int dspal_tester_test_dspal_get_version_info()
 	dspal_get_version_info((char *)version_string, (char *)build_date_string,
 			       (char *)build_time_string);
-
+#endif
```
The LOG_INFO should not print below if dspal_get_version_info is not called. The values will be junk. Something like

LOG_INFO("dspal_get_version_info is currently supported on ADSP only");

mcharleb commented on test/dspal_tester/adsp_proc/pwm_test_imp.c:
```diff
@@ -71,7 +71,7 @@ int dspal_tester_pwm_test(void)
 
-	dspal_get_version_info_ext(&version);
+	//dspal_get_version_info_ext(&version);
 	strtok(version.version_string, "-.");
```
The above line will be junk with dspal_get_version_info_ext commented out.

mcharleb commented on test/include/platform.h:
```diff
@@ -0,0 +1,55 @@
+/****************************************************************************
+ *   Copyright (c) 2015 Mark Charlebois. All rights reserved.
```
You can put your copyright

mcharleb commented on test/version_test/adsp_proc/version_test.c:
```diff
@@ -35,9 +35,10 @@
 
 int version_test_get_version_info(char* version_string, int version_stringLen, char* build_date_string, int build_date_stringLen, char* build_time_string, int build_time_stringLen)
 {
+	#if !defined(DSP_TYPE_SLPI)  // to be enabled after the API is added for SLPI
```
Don't indent preprocessor statements

wangxdflight committed:
updated based on review comments

mcharleb committed:
Added dspal_version_types.h

Changes to cmake_hexagon made it impossible to pass a single
header file using --include

Signed-off-by: Mark Charlebois <mcharleb@qti.qualcomm.com>

wangxdflight committed:
rebased to the lastest

mcharleb commented:
@wangxdflight This is now obsolete. See https://github.com/ATLFlight/dspal/pull/9
